websocket_test.go

```diff
@@ -52,3 +52,286 @@ func TestWebsocketNil(t *testing.T) {
 
 	ws.chain.assertFailed(t)
 }
+
+func TestWebsocketExpect(t *testing.T) {
+
+	t.Run("websocket is closed", func(t *testing.T) {
+		r := newStringReporter()
+		c := &Websocket{
+			chain:    makeChain(r),
+			conn:     &websocket.Conn{},
+			isClosed: true,
+		}
+
+		c.Expect()
+
+		if !r.reported {
+			t.Errorf("Websocket.CloseWithText() error message not reported")
+		}
+
+		want := "\nunexpected read from closed WebSocket connection"
+
+		if got := r.msg; got != want {
```

Comment:
Here and in all tests below. I'd like to avoid matching exact error messages in unit tests, since it would be a burden to update tests when changing formatting, wording, etc. I suggest to check only that the assertion is failed and don't bother with the assertion text.